Modules/_asynciomodule.c

```diff
@@ -2552,6 +2615,9 @@ _asyncio_Task_set_name(TaskObj *self, PyObject *value)
 static void
 TaskObj_finalize(TaskObj *task)
```

Comment:
I guess it is implied from the name and signature of this function, but is `TaskObj_finalize` called only on "native asyncio tasks"? who is responsible to "finalize" non-asyncio tasks? is it the responsibility of the 3rd party task provider to do their own "finalization" and call `_unregister_task`?

answering my own question - this is covered [in the docs](https://docs.python.org/3/library/asyncio-extending.html#task-lifetime-support) - perhaps for clarity a link to the docs can be added around the register / unregister functions

Reply:
Yes you are right, third party implementations needs to call this unregister function as documented themselves to finalize the task. I'll add the link as you suggested, thanks

Reply:
Added link with a comment.